### PR TITLE
surya: Add Xiaomi Poco X3 NFC installer config

### PR DESCRIPTION
**Pull request by ungeskriptet**

### CODE REVIEW

**Comment by amartinz** [comment]
Does the device already have a new recovery, which sets `ro.ubuntu.recovery`?

If it has, you should set:
```
      - actions:
          - systemimage:install:
              verify_recovery: true
```
Reference: https://github.com/ubports/installer-configs/blob/c5134298e2002f7284fc6b2663435971e43b5d54/v2/devices/axolotl.yml#L210

and `compatible_installer` needs to be bumped: `compatible_installer: ">=0.9.4-beta"`

It it has not, it is highly recommended to do so :)

File: v2/devices/surya.yml [hunk 1]
```diff
@@ -0,0 +1,132 @@
+name: "Xiaomi Redmi Note 9 Pro/Pro Max/9S - Poco M2 Pro"
```
Comment (amartinz) [inline v2/devices/surya.yml:1]: Shouldn't this be `Xiaomi POCO X3 NFC`?

File: v2/devices/surya.yml [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    title: "Reboot to Recovery"
+    description: "With the device powered off, hold Volume Up + Power."
+    image: "phone_power_up"
+    button: true
+  bootloader:
+    title: "Reboot to Bootloader"
+    description: "With the device powered off, hold Volume Down + Power."
+    image: "phone_power_down"
+    button: true
+  confirm_model:
```
Comment (amartinz) [inline v2/devices/surya.yml:17]: You should be able to drop this, unless the same device has multiple models and not all of them are supported.

File: v2/devices/surya.yml [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  confirm_firmware:
+    title: "Confirm your firmware"
+    description: "Please double-check that your device is running the latest available stock Android 10 firmware."
+    link: "https://github.com/ubuntu-touch-surya/ubuntu-touch-surya/blob/master/FIRMWARE.md"
+  unlock:
+    title: "OEM unlock"
+    description: "If you haven't done so already, make sure to OEM unlock your device first."
+    link: "https://en.miui.com/unlock/"
+unlock:
+  - "confirm_model"
```
Comment (amartinz) [inline v2/devices/surya.yml:29]: same here, if you drop the above user action.

File: v2/devices/surya.yml (hunk 1 above)
Comment (ungeskriptet) [reply v2/devices/surya.yml:1]: Right, my bad. Will fix this later.

File: v2/devices/surya.yml (hunk 2 above)
Comment (ungeskriptet) [reply v2/devices/surya.yml:17]: I assume you only mean `confirm_model:`?

Comment (amartinz) [reply v2/devices/surya.yml]: Yes, the `confirm_model` section.

There was also a discussion regarding the unlocking of devices.
https://github.com/ubports/installer-configs/commit/54f34fe1a2952c527cdb9407be50a5fdd78d9ed1

Basically you can drop the bootloader unlock user_action as well and if you need a code, you can set `code_url` like here: https://github.com/ubports/installer-configs/blob/54f34fe1a2952c527cdb9407be50a5fdd78d9ed1/v2/devices/FP3.yml#L31

**Review by amartinz — APPROVED** [review]
LGTM, but someone else must hit merge.

**Comment by amartinz** [comment]
@Flohack74 could you have a look, please?

**Review by Flohack74 — APPROVED** [review]

**Pull request merged**